pkg/testmachinery/testflow/validation.go

```diff
@@ -99,7 +107,45 @@ func Validate(identifier string, tf tmv1beta1.TestFlow, locs locations.Locations
 	return nil
 }
 
-func checkIfDependentStepsExist(identifier string, tf tmv1beta1.TestFlow, usedStepNames map[string]bool) error {
+func ensureArtifactsFromStepsExist(identifier string, tf tmv1beta1.TestFlow, usedStepNames map[string]*tmv1beta1.DAGStep) error {
+	for stepName, step := range usedStepNames {
+		identifier := fmt.Sprintf("%s.[%s]", identifier, stepName)
+		if step.ArtifactsFrom != "" {
+			if !previousStepExists(step, step.ArtifactsFrom, usedStepNames) {
+				return fmt.Errorf("%s.artifactsFrom: Invalid value: Step %s is unknown", identifier, step.ArtifactsFrom)
+			}
+		}
+	}
+	return nil
+}
+
+func previousStepExists(step *tmv1beta1.DAGStep, previousStepName string, usedStepNames map[string]*tmv1beta1.DAGStep) bool {
+	// no need to check if ArtifactsFrom is empty
+	if step.ArtifactsFrom == "" {
+		return false
+	}
+
+	if step.DependsOn == nil || len(step.DependsOn) == 0 {
+		// if there are no dependents, then there can be no matching ArtifactsFrom step
+		return false
+	} else {
```

Comment:
no need for the else branch as the condition before returns if true.
So just remove it

Reply:
done